src/Compilers/Core/Portable/SourceGeneration/AdditionalSourcesCollection.cs

```diff
@@ -22,6 +23,10 @@ internal sealed class AdditionalSourcesCollection
 
         private static readonly StringComparer s_hintNameComparer = StringComparer.OrdinalIgnoreCase;
 
+        // Matches relative path segments ("../", "./", "//"), "/" at the beginning,
```

Comment:
>relative path segments ("../", "./", "//"), "/" [](http://example.com/codeflow?start=19&length=47)

Are the parentheses needed (what does the group of three represent)? And it looks like the last two, `"//"` and `"/"` are absolute paths rather than relative.
Perhaps: `// Matches path segments "../", "./", "//", "/" at the beginning,`

Reply:
Only "/" is matched at the beginning, the group ("../", "./", "//") is matched anywhere in the path. I guess I should make that clearer.